### ISSUE TITLE
[CoE Starter Kit - BUG] Attachment missing in Mitigation Plan

### ISSUE THREAD

**Issue by MartinWGT** [microsoft/coe-starter-kit#7056]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

When adding an attachment to the Developer Compliance Center and selecting the 'Mitigation Plan Provided' = Yes, the attachment disappears. Looking in the app 'Power Platform Admin View' the attachment is also missing.

Before saving...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/bac5e902-2660-44d7-92b6-5cca67d924d5)

After saving...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/fd1157eb-8602-48f9-bfd7-fd59e4e695f9)

Admin View app...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/791acb61-572c-4e73-a2cb-e86d29893864)

### Expected Behavior

When an attachment is saved in the DCC support details it should be accessible somewhere.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.16

### What app or flow are you having the issue with?

Developer Compliance Center

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

Open the DCC
Select an app
Click on the Support Details
Enter details and attach a file
Click Save
Click on Support details again
The attachment is missing
Open the 'Power Platform Admin View'  app
Find the app that was updated above
Click on the Governance tab
There is no sign of the attachment

AB#774

**Comment by Jenefer-Monroe** [comment]
I dont repro but this may be a bug with the SR. Can you please tell me which SR the user has? That will help me test.

**Comment by RahulBojji** [comment]
i have tried validating it Power Platform Maker SR but couldn't repro it. Added an attachment while creating the support details. It saved successfully. 

On Developer Compliance Center

<img width="806" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/149372606/5ec60fd3-b1d1-462f-9441-36d3d52a4357">

on Power Platform Admin View
<img width="786" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/149372606/9fb5563d-ef92-4585-a26d-817156fde104">

**Comment by MartinWGT** [comment]
I've just tried it again. I reinstalled CoE in the last few days and am now using the November release.
Unless, I've installed it incorrectly, I'm not able to save anything at all. I can't save the text fields, it's as if both the Save and Cancel button do nothing.

**Comment by MartinWGT** [comment]
I tried editing the DDC to see if I could see why the buttons don't work.
![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/62ac0b63-fb81-427e-bad8-fabd6ef8e4e6)

At first it popped up a message about the Creator Kit components being out of date, but I already have the latest version...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/96bf01a3-38cf-40fc-ae43-f4af81a90e69)


I clicked on the stethoscope icon to see errors and it showed this...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/3bdac678-8ef8-4c3d-b3bb-9f371bd6fd1c)

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/420fd795-01e6-485f-9413-da2d757c37bd)

**Comment by MartinWGT** [comment]
I'm using version CenterofExcellenceAuditComponents_3.13_managed. This is the same as the October release, but I have a new version of Core for November (4.17). Is it correct that the Audit solution was not updated for November?

**Comment by MartinWGT** [comment]
It also says I'm not using the latest version, but refresh does not change anything...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/d0886310-38af-47c7-8c9c-b6b7aa4be5a3)

**Comment by Jenefer-Monroe** [comment]
@MartinWGT can you please post your inability to edit the app as a separate issue?

**Comment by MartinWGT** [comment]
Will do; I was thinking this is a separate problem.

**Comment by RahulBojji** [comment]
@MartinWGT we were able to repro the issue. we will fix it in the upcoming release.
Thanks

**Comment by MartinWGT** [comment]
I'm glad you were able to reproduce this.
It's a bit of a show-stopper though. Can it be fixed before a month's time?

**Comment by MartinWGT** [comment]
Just realised I got my issues mixed up. This can wait until December, although does cause some delays on our end.

**Comment by MartinWGT** [comment]
Interesting, today it is working (see related issue https://github.com/microsoft/coe-starter-kit/issues/7106#issuecomment-1814150088 )

**Comment by MartinWGT** [comment]
Update - it works now for an admin account, but still doesn't work for a Maker with the SR role.

**Comment by RahulBojji** [comment]
@MartinWGT Maker SR doesn't have Append Privileges on Notes (Attachments) table. So, because of these, attachments are not getting created and when you open support details again it doesn't show because it's not created.

**Comment by Jenefer-Monroe** [comment]
Here is the fix to the SR if you'd like to patch ahead of the Dec release



![Image](https://github.com/microsoft/coe-starter-kit/assets/89584709/f6256d39-59f6-4f63-a698-c104719f530b)

**Comment by Jenefer-Monroe** [comment]
Thanks to @RahulBojji for the great investigation work!

**Comment by Jenefer-Monroe** [comment]
also fixed for Admin and User SR

**Comment by MartinWGT** [comment]
Thanks @Jenefer-Monroe . I've modified the SR. Looks like it might take a while to work as no change yet.

**Comment by MartinWGT** [comment]
Hi @Jenefer-Monroe & @RajeevPentyala  It's a day later and changing the SR role hasn't fixed the issue for Makers. I tried in InPrivate mode too.
This is my setting, could there be another permissions that needs changing?

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/768f6a58-061b-49d3-8c1e-4be6240bab9a)

**Comment by RahulBojji** [comment]
@MartinWGT Thanks for highlighting the issue.
<img width="958" alt="MicrosoftTeams-image" src="https://github.com/microsoft/coe-starter-kit/assets/149372606/f29807c0-3880-4e78-8aa1-9012b544e6fd">

There is one more table we need to provide permission for maker SR. Please find it attached above. We have to provide **Read** permission on **SystemJob** table
 You can do it manually at your end to get unblocked or will get it fixed and push this in the upcoming release
Please let us know if you are still facing the issue after the above fix.

**Comment by MartinWGT** [comment]
Thanks @RahulBojji I can't see those permissions for Power Platform Maker SR...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/7512edff-38f5-428d-a61c-4be8e1125a06)

**Comment by Jenefer-Monroe** [comment]
The new UX takes some getting used to :)
You should be able to find as shown here, please do let us know.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/1134afba-6edf-44e0-90c6-684c776a571b)

**Comment by MartinWGT** [comment]
Hi Jenefer,
None of those System tables appear. There are a total of 13 permissions in the Customization table...

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/4fe32c2b-7316-4538-b9a5-b5b2e26e8b7a)

**Comment by RahulBojji** [comment]
Hi @MartinWGT 
Please click the below dropdown and select "Show All tables"

<img width="778" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/149372606/91a8641c-c7bc-4391-8e07-cab8231b58d3">

**Comment by MartinWGT** [comment]
Thanks @RahulBojji 
Got there. That wasn't obvious at all. Microsoft need to revisit their new UI. The only way to see "Show all tables" is to collapse the 'Details' section

Expanded:
![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/368fecf9-3f5c-4026-b147-f32c666a3d70)

Collapsed:
![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/d668bc57-2754-4f3e-8b45-a692e4dc399e)

It's working now, the Maker can save the attachment. 
Thank you!

**Comment by MartinWGT** [comment]
Actually, I can see now - the 'show all tables' is half hidden. I'm using Chrome.

![image](https://github.com/microsoft/coe-starter-kit/assets/142497418/d22ff075-9559-41d3-8c3c-851436085a2f)

**Comment by CoEStarterKitBot** [comment]
@MartinWGT This has been fixed in the latest release. Please install [the latest version of the toolkit](https://aka.ms/CoEStarterKitDownload) following the instructions for [installing updates](https://docs.microsoft.com/power-platform/guidance/coe/setup#installing-updates). Note that if you do not remove the unmanaged layers as described there you will not receive updates from us.
